nupic/encoders/random_distributed_scalar.py

```diff
@@ -265,12 +284,12 @@ def _newRepresentation(self, index, newIndex):
     ri = newIndex % self.w
 
     # Now we choose a bit such that the overlap rules are satisfied.
-    newBit = self.random.randint(self.n)
+    newBit = self.random.getUInt32(self.n)
```

Comment:
Please update __setstate__ so that old checkpoints have `self.random` set to an instance of the NuPIC Random class. This is necessary for upgrade path from old serialization to new (load from old format and then write to new format).